Notebooks/Pluto/p5.jl

```diff
@@ -0,0 +1,359 @@
+
+# ╔═╡ 8cca417e-1dd4-11eb-2a18-bf9b80da3fd2
+md"Lets try the above in Yao! The $ R_𝜑^Z $ gate can be used in Yao with the shift *block*. "
+
+# ╔═╡ 1f7c993e-1dd5-11eb-2f13-7711d2426eb1
+begin
+	qubit = rand_state(1)
+	state(qubit)
+end
+
+# ╔═╡ 5960e486-1dd5-11eb-2253-838f59b06d4c
+state(qubit |> chain(1, put(1=>shift(π/2))))
+
+# ╔═╡ ac10f6f8-1dd5-11eb-2cf1-29480480572a
+md"As expected the output was $\begin{bmatrix}a\\ib\end{bmatrix}$. Remember, $ i = \sqrt{-1} $ and $ i^2 = -1 $. (Also, note that in Julia, imaginary number 𝑖 is represented by im.)"
+
+# ╔═╡ af22f8a8-1ddc-11eb-2958-bb45cdadf44f
+md"Also, $ R_𝜋^Z $ gate is equivalent to Z gate."
+
+# ╔═╡ df5a774e-1ddc-11eb-2e86-1dc8dd502e26
+round.(Matrix(chain(1, put(1=>shift(π))))) == round.(Matrix(chain(1, put(1=>Z)))) #The round functions "rounds-off" the elements of the matrices
+
+# ╔═╡ c1f87994-1e7c-11eb-3da3-21aa52e80520
+md"Its represented in a circuit diagram by,"
+
+# ╔═╡ cda277ce-1e7c-11eb-246c-0b97bb66ab73
+plot(chain(1, put(1=>shift(π/3))))
+
+# ╔═╡ a1173196-1dd9-11eb-394d-271a3871e181
+md"#### The T Gate"
+
+# ╔═╡ b1cc417a-1dd9-11eb-0156-67fd10743000
+md"The T gate is equivalent to $ R_\frac{𝞹}{4}^Z . $ In its matrix form, it can be written as $\begin{bmatrix}1 & 0\\0 & \frac{1 + i}{\sqrt{2}}\end{bmatrix}$. Nevertheless, in Yao, it can be used by using the **T** *block* ."
+
+# ╔═╡ 67b2bc30-1dda-11eb-2d81-edd411a207b2
+state(qubit |> chain(1, put(1=>T)))
+
+# ╔═╡ 7d546e10-1dda-11eb-2b9a-abe61e9edf6c
+md"Also,"
+
+# ╔═╡ 83dfc128-1dda-11eb-37d9-37b959ea5f58
+Matrix(chain(1, put(1=>shift(π/4)))) == Matrix(chain(1, put(1=>T)))
+
+# ╔═╡ ea6c8192-1e7c-11eb-30e5-8d65801ce703
+md"Its circuit diagram representation looks somewhat like - "
+
+# ╔═╡ feab87f0-1e7c-11eb-075b-a55e9d86184c
+plot(chain(1, put(1=>T)))
+
+# ╔═╡ 408ce964-1de4-11eb-2830-f14bdc87f899
+md"#### The $ R_𝜑^X $ gate"
+
+# ╔═╡ 09a069f0-1df6-11eb-19de-5da4bb572510
+md"Similar to the $ R\_𝜑^Z $ gate, the $ R\_𝜑^X $ gate can be represented by $\begin{bmatrix}cos(\frac{𝜑}{2}) & -sin(\frac{𝜑}{2})i\\-sin(\frac{𝜑}{2})i & cos(\frac{𝜑}{2})\end{bmatrix}$."
+
+# ╔═╡ 5a6f6786-1df7-11eb-0f06-5b229b0935bb
+md"#### The $ R_𝜑^Y $ gate"
+
+# ╔═╡ 6a775062-1df7-11eb-1ea2-7d5517774328
+md"Similar to the $ R\_𝜑^Z $ gate, the $ R\_𝜑^X $ gate can be represented by $\begin{bmatrix}cos(\frac{𝜑}{2}) & -sin(\frac{𝜑}{2})\\sin(\frac{𝜑}{2}) & cos(\frac{𝜑}{2})\end{bmatrix}$."
+
+# ╔═╡ 8d65504c-1df7-11eb-0f3a-93796bf5b7f9
+md"They can be represented in Yao using the **Rx** and **Ry** *blocks* respectively" 
+
+# ╔═╡ e10cd2c0-1e00-11eb-3c86-2d958e7aa92f
+state(qubit |> chain(1, put(1=>Rx(π))))
+
+# ╔═╡ 38071b04-1e00-11eb-1c6c-03f03ce37a92
+state(qubit |> chain(1, put(1=>Ry(π))))
+
+# ╔═╡ 451b4c6a-1e1f-11eb-18f0-e301d5d34c93
+md"There's also an **Rz** *block* which which represents the alternative form of $ R_𝜑^Z $ matrix, i.e., $\begin{bmatrix}e^{-𝜑i/2} & 0\\0 & e^{𝜑i/2}\end{bmatrix}$."
```

Comment:
maybe would be better if we print the docstring of `Rz` etc. directly? so people can learn where to look for docs.

Reply:
Check out the README.md 
This part's refers to the docs！

Reply:
Also, if they open it in Pluto, isn't the docs already there？